java-util/src/main/java/io/druid/java/util/common/granularity/AllGranularity.java

```diff
@@ -26,6 +26,7 @@
 
 /**
  * AllGranularty buckets everything into a single bucket
+ * This class is serialized and deserialized based on type in GranularityModule
```

Comment:
Could you please move this statement to the constructor javadoc and add "this constructor is public because ..."